wagtail/admin/views/generic/base.py

```diff
@@ -364,76 +325,11 @@ def active_filters(self):
-                                field_name, item.pk
-                            ),
-                        )
-                    )
-            elif isinstance(filter_def, MultipleChoiceFilter):
-                choices = flatten_choices(filter_def.field.choices)
-                for item in value:
-                    filters.append(
-                        ActiveFilter(
-                            bound_field.auto_id,
-                            filter_def.label,
-                            choices.get(str(item), str(item)),
-                            self.get_url_without_filter_param_value(field_name, item),
-                        )
-                    )
-            elif isinstance(filter_def, ModelChoiceFilter):
-                field = filter_def.field
-                filters.append(
-                    ActiveFilter(
-                        bound_field.auto_id,
-                        filter_def.label,
-                        field.label_from_instance(value),
-                        self.get_url_without_filter_param(field_name),
-                    )
-                )
-            elif isinstance(filter_def, DateFromToRangeFilter):
-                start_date_display = date_format(value.start) if value.start else ""
-                end_date_display = date_format(value.stop) if value.stop else ""
-                widget = filter_def.field.widget
-                filters.append(
-                    ActiveFilter(
-                        bound_field.auto_id,
-                        filter_def.label,
-                        f"{start_date_display} - {end_date_display}",
-                        self.get_url_without_filter_param(
-                            [
-                                widget.suffixed(field_name, suffix)
-                                for suffix in widget.suffixes
-                            ]
-                        ),
-                    )
-                )
-            elif isinstance(filter_def, ChoiceFilter):
-                choices = flatten_choices(filter_def.field.choices)
-                filters.append(
-                    ActiveFilter(
-                        bound_field.auto_id,
-                        filter_def.label,
-                        choices.get(str(value), str(value)),
-                        self.get_url_without_filter_param(field_name),
-                    )
-                )
-            else:
-                filters.append(
-                    ActiveFilter(
-                        bound_field.auto_id,
-                        filter_def.label,
-                        str(value),
-                        self.get_url_without_filter_param(field_name),
-                    )
-                )
+            filter_adapter_class = filter_adapter_class_registry.get(filter_def)
+            filter_adapter = filter_adapter_class(
+                filter_def, bound_field, value, base_url, query_dict
+            )
```

Comment:
We don't pass `field_name`, as it can be retrieved from `bound_field.name` instead. I think they're both equal...